shapely/algorithms/polylabel.py

```diff
@@ -0,0 +1,115 @@
+from ..geometry import Point, LineString
+from ..geos import TopologicalError
+from heapq import heappush, heappop
+
+
+class Cell(object):
+    def __init__(self, x, y, h, polygon):
+        self.centroid = Point(x, y)  # cell centroid
+        self.h = h  # half of cell size
+
+        # distance from cell centroid to polygon exterior
+        self.distance = self._dist(polygon)
+
+        # max distance to polygon within a cell
+        self.max_distance = self.distance + h * 1.4142135623730951  # sqrt(2)
+
+    # rich comparison operators for sorting in minimum priority queue
+    def __lt__(self, other):
+        # inverted for minimum priority queue
+        if self.max_distance < other.max_distance:
+            return False
+        return True
```

Comment:
@allanadair how about something like `return self.max_distance >= other.max_distance` instead, and similarly in the other methods?